server/src/main/java/io/crate/planner/consumer/CreateTableAsPlanner.java

```diff
@@ -49,14 +51,16 @@ public static Plan plan(AnalyzedCreateTableAs analyzedCreateTableAs,
                             TableCreator tableCreator,
                             Schemas schemas,
                             PlannerContext context,
-                            LogicalPlanner logicalPlanner) {
+                            LogicalPlanner logicalPlanner,
+                            ClusterService clusterService) {
```

Comment:
`execute` of a plan can access the current cluster state via the planner context, so I think this may not be necessary. 

It would help ensure that the full planning phase is based on a consistent cluster state view.